plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -480,51 +479,29 @@ public Map<String, Object> getTableProperties(ConnectorSession session, JdbcTabl
     }
 
     public Map<String, Object> getTableProperties(Connection connection, JdbcTableHandle tableHandle)
+            throws SQLException
     {
-        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
         SchemaTableName schemaTableName = tableHandle.asPlainTable().getSchemaTableName();
+        String schemaName = requireNonNull(schemaTableName.getSchemaName(), "Ignite schema name can not be null").toUpperCase(ENGLISH);
         String tableName = requireNonNull(schemaTableName.getTableName(), "Ignite table name can not be null").toUpperCase(ENGLISH);
-        String sql = "SELECT idx.CACHE_ID, " +
-                "(SELECT COLUMNS FROM sys.indexes WHERE SCHEMA_NAME = 'PUBLIC' AND TABLE_NAME = ? AND INDEX_NAME = '_key_PK') AS PKS," +
-                "(SELECT COLUMNS FROM sys.indexes WHERE SCHEMA_NAME = 'PUBLIC' AND TABLE_NAME = ? AND INDEX_NAME = 'AFFINITY_KEY') AS AFK FROM sys.indexes as idx " +
-                "JOIN sys.caches che ON idx.CACHE_ID = che.CACHE_ID WHERE idx.SCHEMA_NAME = 'PUBLIC' AND idx.TABLE_NAME = ? LIMIT 1";
-
-        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
-            preparedStatement.setString(1, tableName);
-            preparedStatement.setString(2, tableName);
-            preparedStatement.setString(3, tableName);
-
-            try (ResultSet resultSet = preparedStatement.executeQuery()) {
-                if (!resultSet.next()) {
-                    return ImmutableMap.of();
-                }
-                List<String> primaryKeys = extractColumnNamesFromOrderingScheme(resultSet.getString("PKS"));
-                checkArgument(!primaryKeys.isEmpty(), "Ignite table should has at least one primary key");
-                properties.put(PRIMARY_KEY_PROPERTY, primaryKeys);
-            }
-        }
-        catch (SQLException e) {
-            throw new TrinoException(JDBC_ERROR, e);
-        }
-        return properties.buildOrThrow();
-    }
 
-    // extract result from format: "ID" ASC, "CITY_ID" ASC
-    private List<String> extractColumnNamesFromOrderingScheme(String row)
-    {
-        ImmutableList.Builder<String> builder = ImmutableList.builder();
-        if (isNullOrEmpty(row)) {
-            return builder.build();
-        }
+        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+        DatabaseMetaData metaData = connection.getMetaData();
+
+        try (ResultSet resultSet = metaData.getPrimaryKeys(null, schemaName, tableName)) {
```

Comment:
This won't work correctly when the table name contains pattern. e.g. `_` 
```sql
trino:public> create table a_c (c1 int, c2 int) with (primary_key = array['c2']);
CREATE TABLE
trino:public> create table abc (c1 int, c2 int) with (primary_key = array['c2']);
CREATE TABLE
trino:public>
trino:public> show create table a_c;
Query 20230303_130523_00034_t9hi4 failed: Multiple tables matched: public.a_c
```
Please fix and add the test case.

Reply:
Hi this is introduced by the implementation `IgniteClient#getTables`. I filed another session to handle it. see 
https://github.com/trinodb/trino/pull/16364

Reply:
Adding #16364 is insufficient.

Reply:
Are you suggest we need to do the escape operations on while request `metaData.getPrimaryKeys`

Reply:
Yes.

Reply:
I think it's maybe unnecessary to wrap the params? because I saw the params description(`DatabaseMetaData#getPrimaryKeys`), they are not pattern:
```     
     * @param catalog a catalog name; must match the catalog name as it
     *        is stored in the database; "" retrieves those without a catalog;
     *        {@code null} means that the catalog name should not be used to narrow
     *        the search
     * @param schema a schema name; must match the schema name
     *        as it is stored in the database; "" retrieves those without a schema;
     *        {@code null} means that the schema name should not be used to narrow
     *        the search
     * @param table a table name; must match the table name as it is stored
     *        in the database
     * @return {@code ResultSet} - each row is a primary key column description
     * @throws SQLException if a database access error occurs
     */
Am I miss sth?

Reply:
> I think it's maybe unnecessary to wrap the params? 

No, it's necessary. 

> Am I missing sth?

Unfortunately, yes. I would recommend looking into Ignite JDBC driver's code.

Reply:
You are right. The field shows they are treated as the pattern, and final handler do translate them to the regex.

Reply:
Unfortunately, we can not pass test `BaseConnectorTest.testReadMetadataWithRelationsConcurrentModifications`, the error shows that sometimes `metaData.getPrimaryKeys` return empty list of primary keys in Ignite, which shouldn't.
Maybe this part we could use the original implementation, one of the pros is that it's easier to add more beneficial property.

Reply:
Reverted. Using raw sql to get primary keys.